e2e/utils.go

```diff
@@ -82,6 +85,133 @@ var (
 	clusterID        string
 )
 
+type cephfsFilesystem struct {
+	Name           string   `json:"name"`
+	MetadataPool   string   `json:"metadata_pool"`
+	MetadataPoolID int      `json:"metadata_pool_id"`
+	DataPoolIDs    []int    `json:"data_pool_ids"`
+	DataPools      []string `json:"data_pools"`
+}
+
+// listCephFSFileSystems list CephFS filesystems in json format.
+func listCephFSFileSystems(f *framework.Framework) ([]cephfsFilesystem, error) {
+	var fsList []cephfsFilesystem
+
+	stdout, stdErr, err := execCommandInToolBoxPod(
+		f,
+		"ceph fs ls --format=json",
+		rookNamespace)
+	if err != nil {
+		return fsList, err
+	}
+	if stdErr != "" {
+		return fsList, fmt.Errorf("error listing fs %v", stdErr)
+	}
+
+	err = json.Unmarshal([]byte(stdout), &fsList)
+	if err != nil {
+		return fsList, err
+	}
+
+	return fsList, nil
+}
+
+// getCephFSPoolName get CephFS pool name from filesystem name.
+func getCephFSPoolName(f *framework.Framework, filesystem string) (string, error) {
+	fsList, err := listCephFSFileSystems(f)
+	if err != nil {
+		return "", err
+	}
+	for _, v := range fsList {
+		if v.Name != filesystem {
+			continue
+		}
+
+		return v.MetadataPool, nil
+	}
+
+	return "", os.ErrNotExist
+}
+
+func compareStdoutWithCount(stdOut string, count int) {
+	stdOut = strings.TrimSuffix(stdOut, "\n")
+	res, err := strconv.Atoi(stdOut)
+	if err != nil {
+		e2elog.Failf("failed to convert string to int: %s", stdOut)
+	}
+	if res != count {
+		e2elog.Logf("expected omap object count %d, got %d", count, res)
+	}
+}
+
+// validateOmapCount validates no of OMAP entries on the given pool.
+// Works with Cephfs and RBD drivers and mode can be snaps or volumes.
+func validateOmapCount(f *framework.Framework, count int, driver, pool, mode string) {
+	var (
+		cmd1   string
+		cmd2   string
```

Comment:
```suggestion
radosLsCmd   string
radosLsKeysCmd   string
```
we can use better naming for the cmds?